compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -578,6 +608,12 @@ abstract class PrepJSInterop extends plugins.PluginComponent
           // Mark module classes as having the new format
           if (sym.isModuleClass)
             sym.addAnnotation(HasJSNativeLoadSpecAnnotation)
+        } else {
+          assert(sym.isTrait) // just tested in the previous `if`
+          if (sym.hasAnnotation(JSImportAnnotation)) {
+            reporter.error(implDef.pos,
+                "Traits may not have an @JSImport annotation")
+          }
```

Comment:
Hmmm... Probably we should also emit a warning for `@JSName` as it does not do anything. Filed as #2617.